### PR TITLE
Add hdfs support.

### PR DESCRIPTION
**Pull request by haosdent**
Configuration is 

```
storage:
  hdfs:
    namenode: 127.0.0.1:8020  # Your hdfs namenode addresses.
    rootdirectory: /registry
```

Based on this library https://github.com/colinmarc/hdfs

Work well in our production, stable then ceph 1024 times.

### CODE REVIEW

File: registry/storage/driver/hdfs/driver.go [hunk 1]
```diff
@@ -0,0 +1,449 @@
+package hdfs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/colinmarc/hdfs"
```
Comment (thaJeztah) [inline registry/storage/driver/hdfs/driver.go]: This import should be in the group below

Also looks like `go.mod` needs to be updated for the new dependency (and vendored)

Comment (haosdent) [reply registry/storage/driver/hdfs/driver.go]: OK

**Comment by dmcgowan** [comment]
@haosdent we aren't accepting any new drivers under the `github.com/docker/distribution/registry/storage/driver/` package right now. If you can put this driver into a separate repository with instructions for how to import it into the main, we will link to it. Only drivers that have proven to be maintained, stable and in high demand will be considered for inclusion in the official builds.

**Comment by milosgajdos** [comment]
Closing as per @dmcgowan comment. No new drivers will be added to this repo. In the future, we will aim to make the drivers pluggable but there is not ETA.

**Pull request closed without merging**